core/types/types.go

```diff
@@ -27,8 +27,9 @@ type Extrinsic []byte
 
 // Block defines a state block
 type Block struct {
-	Header BlockHeader
-	Body   BlockBody
+	Header      BlockHeader
+	Body        BlockBody
+	ArrivalTime uint64 // arrival time of this block
```

Comment:
please change ArrivalTime to an unexported field (change to `arrivalTime`) and add a getter/setter for it.

this is because when we encode the block, we want it to just have `Header` and `Body`, unexporting the field allows the encoder to ignore it.